modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,296 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import type Editor from 'tinymce/core/api/Editor';
+import type { User } from 'tinymce/core/lookup/UserLookup';
+
+const createMockUser = (id: string): User => ({
+  id,
+  name: 'Test User',
+  avatar: 'test-avatar.png',
+  description: 'Test Description'
+});
+
+describe('browser.tinymce.core.UserLookupTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    current_user_id: 'test-user-1',
+    fetch_users: (userIds: string[]): Promise<User[]> => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const users = Arr.bind(userIds, (id) =>
+            id.startsWith('non-existent') ? [] : [ createMockUser(id) ]
+          );
+
+          resolve(users);
+        }, 0);
+      });
+    },
+  });
+
+  it('Should return the configured user ID', () => {
+    const editor = hook.editor();
+    const currentUserId = editor.userLookup.getCurrentUserId();
+    assert.equal(currentUserId, 'test-user-1', 'Should return the configured user ID');
+  });
+
+  it('Should fetch users and return array of promises', () => {
+    const editor = hook.editor();
+    const userIds = [ 'test-user-1' ];
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+
+    assert.lengthOf(promises, 1, 'Should return array with one promise');
+    assert.isTrue(promises[0] instanceof Promise, 'Should return a Promise');
+  });
+
+  it('Should return multiple promises for multiple userIds', () => {
+    const editor = hook.editor();
+    const userIds = [ 'test-user-1', 'test-user-2' ];
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+
+    assert.lengthOf(promises, 2, 'Should return array with two promises');
+    assert.isTrue(promises[0] instanceof Promise, 'First item should be a Promise');
+    assert.isTrue(promises[1] instanceof Promise, 'Second item should be a Promise');
+  });
+
+  it('Should throw an error for empty string ids', async () => {
+    const editor = hook.editor();
+    const userIds = [ '' ];
+    try {
+      await Promise.all(editor.userLookup.fetchUsers(userIds));
+      assert.fail('Should throw error for empty string ID');
+    } catch (error) {
+      assert.isDefined(error, 'Should handle empty string ID with error');
+    }
```

Comment:
We have `Chai` available, I think it has helper methods for "should throw an error"? I'm not sure if we have existing tests to pull examples from.